Bulldozer.CSV/Maps/GroupMemberHistorical.cs

```diff
@@ -0,0 +1,422 @@
+                        GroupMember = groupMember,
+                        GroupMemberId = groupMember.Id,
+                        GroupMemberHistoricalForeignKey = string.Format( "{0}^{1}_{2}_{3}", ImportInstanceFKPrefix, groupMemberHistoricalCsv.GroupId, groupMemberHistoricalCsv.PersonId, long.Parse( groupMemberHistoricalCsv.EffectiveDateTime.ToString( "yyMMddHHmm" ) ) ),
+                        GroupTypeId = groupMember.GroupTypeId,
+                        PersonId = groupMember.PersonId,
+                        InactiveDateTime = groupMemberHistoricalCsv.InactiveDateTime,
+                        IsArchived = groupMemberHistoricalCsv.IsArchived,
+                        IsLeader = groupMemberHistoricalCsv.IsLeader,
+                        Role = groupMemberHistoricalCsv.Role.Left( 100 ),
+                    };
+                    if ( groupMemberHistoricalCsv.GroupMemberStatusHistorical.HasValue )
+                    {
+                        newGroupMemberHistorical.GroupMemberStatus = groupMemberHistoricalCsv.GroupMemberStatusHistorical.Value;
+                    }
+                    groupMemberHistoricalImports.Add( newGroupMemberHistorical );
+                }
+            }
+
+            var groupMemberHistorical = groupMemberHistoricalImports.Where( v => !groupMemHistLookup.ContainsKey( v.GroupMemberHistoricalForeignKey ) ).ToList();
+
+            // Add GroupType Roles if needed
+
+            BulkInsertMemberHistoricalGroupTypeRoles( rockContext, groupMemberHistorical, groupTypeRoleLookup );
+
+            var groupMemberHistoricalToInsert = new List<GroupMemberHistorical>();
+
+            var groupMemHistImportByGroupType = groupMemberHistorical.GroupBy( a => a.GroupTypeId.Value )
+                                                .Select( a => new
+                                                {
+                                                    GroupTypeId = a.Key,
+                                                    GroupMemberHistoricals = a.Select( x => x ).ToList()
+                                                } );
+
+            // populate GroupMemberHistorical records
+            foreach ( var groupMemHistImportObj in groupMemHistImportByGroupType )
+            {
+                var groupTypeRoleNameLookup = groupTypeRoleLookup.GetValueOrNull( groupMemHistImportObj.GroupTypeId ).ToDictionary( k => k.Name, v => v );
+
+                foreach ( var groupMemHistImport in groupMemHistImportObj.GroupMemberHistoricals )
+                {
+                    var groupRole = groupTypeRoleNameLookup.GetValueOrNull( groupMemHistImport.Role );
+                    if ( groupRole == null )
+                    {
+                        groupMHErrors += $"{DateTime.Now}, GroupMember, Group Role {groupMemHistImport.Role} not found in Group Type. Group Member for Rock GroupId {groupMemHistImport.GroupId}, Rock PersonId {groupMemHistImport.PersonId} was set to default group type role.\"Member\".\r\n";
```

Comment:
Convert this to capture error but still create historical entry, setting role to default group type role.